tests/regression_tests/surface_source_write/test.py

```diff
@@ -0,0 +1,991 @@
+    water.set_density("g/cc", 1.0)
+    water.add_s_alpha_beta("c_H_in_H2O")
+    water.id = 41
+
+    materials = openmc.Materials([u235, water])
+    model.materials = materials
+
+    # =============================================================================
+    # Geometry
+    # =============================================================================
+
+    dagmc_univ = openmc.DAGMCUniverse(Path("../dagmc.h5m"))
+    model.geometry = openmc.Geometry(dagmc_univ)
+
+    # =============================================================================
+    # Settings
+    # =============================================================================
+
+    model.settings = openmc.Settings()
+    model.settings.particles = 100
+    model.settings.batches = 5
+    model.settings.inactive = 1
+    model.settings.seed = 1
+
+    source_box = openmc.stats.Box([-4, -4, -20], [4, 4, 20], only_fissionable=True)
+    model.settings.source = openmc.IndependentSource(space=source_box)
+
+    return model
+
+
+@pytest.fixture(scope="module")
+def model_dagmc_2():
+    """Model based on the mesh file 'dagmc.h5m' available from
+    tests/regression_tests/dagmc/legacy.
+
+    This model corresponds to the model_dagmc_1 contained in two boxes to introduce
+    multiple level of coordinates from CSG geometry.
+
+    """
+    openmc.reset_auto_ids()
+    model = openmc.Model()
+
+    # =============================================================================
+    # Materials
+    # =============================================================================
+
+    u235 = openmc.Material(name="no-void fuel")
+    u235.add_nuclide("U235", 1.0, "ao")
+    u235.set_density("g/cc", 11)
+    u235.id = 40
+
+    water = openmc.Material(name="water")
+    water.add_nuclide("H1", 2.0, "ao")
+    water.add_nuclide("O16", 1.0, "ao")
+    water.set_density("g/cc", 1.0)
+    water.add_s_alpha_beta("c_H_in_H2O")
+    water.id = 41
+
+    materials = openmc.Materials([u235, water])
+    model.materials = materials
+
+    # =============================================================================
+    # Geometry
+    # =============================================================================
+
+    dagmc_univ = openmc.DAGMCUniverse(Path("../dagmc.h5m"))
+
+    # -----------------------------------------------------------------------------
+    # Box 1
+    # -----------------------------------------------------------------------------
+
+    # Parameters
+    box1_size = 44
+
+    # Surfaces
+    box1_lower_plane = openmc.ZPlane(z0=-box1_size / 2.0, surface_id=101)
+    box1_upper_plane = openmc.ZPlane(z0=box1_size / 2.0, surface_id=102)
+    box1_left_plane = openmc.XPlane(x0=-box1_size / 2.0, surface_id=103)
+    box1_right_plane = openmc.XPlane(x0=box1_size / 2.0, surface_id=104)
+    box1_rear_plane = openmc.YPlane(y0=-box1_size / 2.0, surface_id=105)
+    box1_front_plane = openmc.YPlane(y0=box1_size / 2.0, surface_id=106)
```

Comment:
Is it necessary to set the surface IDs manually or can these be removed?

Reply:
I need to set the surface IDs here to avoid using the same IDs in the DAGMC universe and in the CSG geometry.